### PR TITLE
Use broccoli-persistent-filter

### PR DESCRIPTION
**Pull request by nickiaconis**
Hot on the heels of #14, here are the changes to use broccoli-persistent-filter. I've set it up to persist by default. It can be overridden by passing `{ persist: false }` as an option.

~~Tests are passing: https://travis-ci.org/nickiaconis/broccoli-eslinter/builds/115196276~~
**Edit:** This repo now leverages its own CI testing.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -91,21 +101,48 @@ EslintValidationFilter.prototype.constructor = EslintValidationFilter;
 EslintValidationFilter.prototype.extensions = ['js'];
 EslintValidationFilter.prototype.targetExtension = 'js';
 
-EslintValidationFilter.prototype.processString = function processString(content, relativePath) {
-  'use strict'; // eslint-disable-line strict
+EslintValidationFilter.prototype.baseDir = function baseDir() {
+  return process.cwd();
```
Comment (rwjblue) [inline lib/index.js]: This is generally __dirname

File: lib/index.js [hunk 2]
```diff
@@ -91,21 +101,48 @@ EslintValidationFilter.prototype.constructor = EslintValidationFilter;
...
+};
+
+EslintValidationFilter.prototype.cacheKeyProcessString = function cacheKeyProcessString(content, relativePath) {
+  return md5Hex([content, relativePath, JSON.stringify(this.options)]);
```
Comment (rwjblue) [inline lib/index.js]: We need to use json-stable-stringify as the unordered json stringify busts the cache incorrectly

Comment (rwjblue) [inline lib/index.js]: We also need to handle the testGenerator function manually here, I think

**Comment by nickiaconis** [comment]
@rwjblue Good points. I've addressed them, and this should be ready for another review.

**Comment by nickiaconis** [comment]
Added the result of `getConfigForFile` to the file cache key, per Robert's [comment on #26](https://github.com/jonathanKingston/broccoli-lint-eslint/issues/26#issuecomment-199570975).

**Comment by BrianSipple** [comment]
@nickiaconis pinging for a heads-up about a few conflicts after merging #25.

**Comment by nickiaconis** [comment]
Thanks for the heads-up. Conflicts resolved.

**Pull request merged**